### DISCUSSION TITLE
A problem with 'ADSoretCoeffTest' in "Misc" Modules

### DISCUSSION THREAD

**Discussion by sinaheidari22** [idaholab/moose#16605] [Q&A Modules: General] [answered]
Hello everyone, 
Happy new year to you, 

I was trying to run 'ad_thermo_diffusion.i'  input file which is available in this address: modules/misc/test/tests/kernels/thermo_diffusion/ad_thermo_diffusion.i

I got this error:

> *** ERROR ***A 'ADSoretCoeffTest' is not a registered object.
> If you are trying to find this object in a dynamically linked library, make sure thatthe library can be found either in your "Problem/library_path" parameter or in the MOOSE_LIBRARY_PATH environment variable.

====================================================

Can you please advise on that, 
Thank you in advance.

**Comment by GiudGiud** [comment]
Hello

Happy New Year to you too!
I think ADSoretCoeffTest is a test object, so a regular moose executable might not be linked to a library that contains it. 
Which executable are you using to run this test? The one built in the misc folder?
If this material object is the one you need, you can just move the source code in modules/misc/test/src/materials/ADSoretCoeffTest.C to modules/misc/src/materials/ADSoretCoeffTest.C, same for the header. Then it will be compiled in the module. 

Best,
Guillaume

EDIT See logan's answer

**Reply by sinaheidari22** [reply]
Thank you for your reply.

**Comment by loganharbour** [comment, marked as answer]
You can use an application’s test objects by passing the `--allow-test-objects` command line flag when you call the executable. 

Test objects are meant to be used in testing only. If you wish to use them in a production environment, we recommend creating your own application and creating an object specific for your needs.

**Reply by sinaheidari22** [reply]
Thank you for your reply, I did like that and had successful run.

**Reply by ZJYZXH** [reply]
Hello, I have such a mistake in the program I created, how should I solve it?

**Reply by GiudGiud** [reply]
Please create your own discussions post. And paste the error message there. If it s the same object missing, it s the same solution Logan wrote down here

**Reply by aaaaaaqing** [reply]
i cant understand ，what should we do
